### PR TITLE
Add Detekt and Github Action to run on Pull Request

### PR DESCRIPTION
**Pull request by scottschmitz**
Setup Detekt and have it run on pull request. We are limited to 5,000 minutes per month of free tier so I set it to only on pull request to prevent getting anywhere close to that - though I'm not sure we would anyway. [It's very easy to change that...](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration)

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Comment (joshfriend) [inline .idea/.gitignore]: Can you pull the `.idea` folder out please? Seems like a good thing to do in a separate PR

**Review by joshfriend — CHANGES_REQUESTED** [review]

File: .idea/.gitignore (hunk 1 above)
Comment (scottschmitz) [reply .idea/.gitignore]: Yea that was unintentional, I'll remove those

Comment (scottschmitz) [reply .idea/.gitignore]: done

Comment (joshfriend) [reply .idea/.gitignore]: cool, we can add it back in #3
